src/components/interactive-builder/add-question-modal.component.tsx

```diff
@@ -580,6 +618,95 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
                   />
                 </RadioButtonGroup>
               ) : null}
+
+              {questionType === 'programState' && (
+                <>
+                  <div>
+                    {isLoadingPrograms && (
+                      <InlineLoading
+                        className={styles.loader}
+                        description={t('searchingPrograms', 'Searching programs') + '...'}
+                      />
+                    )}
+                    {programsLookupError ? (
+                      <InlineNotification
+                        kind="error"
+                        lowContrast
+                        className={styles.error}
+                        title={t('errorFetchingPrograms', 'Error fetching programs')}
+                        subtitle={t('pleaseTryAgain', 'Please try again.')}
+                      />
+                    ) : null}
+                    {programs && (
+                      <ComboBox
+                        helperText={t('programHelperText', 'Patient program field')}
+                        id="programLookup"
+                        items={programs}
+                        itemToString={(item: Program) => item?.name}
+                        onChange={({ selectedItem }) => {
+                          handleProgramChange(selectedItem as Program);
+                        }}
+                        placeholder={t('addProgram', 'Add program')}
+                        selectedItem={selectedProgram}
+                      />
+                    )}
+
+                    {selectedProgram && (
+                      <ComboBox
+                        helperText={t(
+                          'programWorkflowHelperText',
+                          'Program workflow fields must be linked to a program',
+                        )}
+                        id="programWorkflowLookup"
+                        items={programWorkflows}
+                        itemToString={(item: ProgramWorkflow) => item?.concept?.display}
+                        onChange={({ selectedItem }) => {
+                          handleProgramWorkflowChange(selectedItem as ProgramWorkflow);
+                        }}
+                        placeholder={t('addProgramWorkflow', 'Add program workflow')}
+                        selectedItem={programWorkflowUuid}
+                        className={styles.programsDropdown}
```

Comment:
```suggestion
											 selectedItem={programWorkflows.find((workflow) => workflow.uuid === programWorkflowUuid)}
                        titleText={t('programWorkflow', 'Program workflow')}
```

1. The `<Stack>` component will apply the appropriate spacing between the fields. We wouldn't need the additional class.
2. Setting `selectedItem` to `programWorkflowUuid` means we won't see any text rendered in the `ComboBox`. We need to find the workflow whose UUID matches `programWorkflowUuid`.